src/applications/Applications.DicomLoader/Loader.cs

```diff
@@ -0,0 +1,165 @@
+﻿#nullable enable
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using DicomTypeTranslation;
+using FellowOakDicom;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using SharpCompress.Readers;
+using Smi.Common.Messages;
+using Smi.Common.MongoDB;
+
+namespace Applications.DicomLoader;
+
+public static class Loader
```

Comment:
Seems like everything in this class is static.  Could we use a singleton pattern or something instead? Static stuff is usually harder to maintain (can't subclass later on, harder to test etc)?